### PR TITLE
[hotfix] set enforce-index-use default value to strict

### LINKED ISSUE #4884: [BUG] enforce-index-use is omitting search results
`enforce-index-use` is a value that is set in `etc/conf.xml`  and documented as follows:
```
 enforce-index-use
               When set to "strict", eXist will not use a range index unless all
                collections in the context sequence define it. When set to
                "always", the query engine will still use an index, even if only
                one collection has it defined. It thus leaves it to the user to
                properly define indexes and if you forget to specify an index on
                a particular collection, it will be missing in the results.
``` 
The value, by default, is set to `always`, and according to the documentation, this means it will ignore any collection without an index defined.
Most users are unaware of this, for eg: #3620. I found it while trying to fix the issue just mentioned. The following is the code that lead me to this discovery: https://github.com/eXist-db/exist/blob/165bbc58a3cf126a1749000f9a794395e80c39c0/exist-core/src/main/java/org/exist/xquery/pragmas/Optimize.java#L342
I don’t think the default value should be `always`. Ignoring search results without users knowledge should not be the default case.
@joewiz @adamretter

### PR DESCRIPTION
**Pull request by marmoure**
Fixes #4884 
As discussed in #4884 we want to make the default value to be `strict`.

---
This open source contribution to the [exist](https://github.com/eXist-db/exist) project was commissioned by the Office of the Historian, U.S. Department of State, https://history.state.gov/.

### CODE REVIEW

**Comment by joewiz** [comment]
Naturally, this PR would require a new major release. Today, this would mean it could be a candidate for eXist 7. So I'll set it for that milestone.

**Comment by adamretter** [comment]
Tests will follow shortly...

File: extensions/indexes/indexes-integration-tests/src/test/java/org/exist/indexing/EnforceIndexUseTest.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+            Collection oldRange = broker.getOrCreateCollection(transaction, XmldbURI.xmldbUriFor(TestConstants.TEST_COLLECTION_URI + "old-range-index"));
+            Collection noRange = broker.getOrCreateCollection(transaction, XmldbURI.xmldbUriFor(TestConstants.TEST_COLLECTION_URI + "no-range-index"));
+            broker.saveCollection(transaction, test);
+            broker.saveCollection(transaction, newRange);
+            broker.saveCollection(transaction, oldRange);
+            broker.saveCollection(transaction, noRange);
+
+            transact.commit(transaction);
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
```
Comment (reinhapa) [inline extensions/indexes/indexes-integration-tests/src/test/java/org/exist/indexing/EnforceIndexUseTest.java:130]: Remove this catch and add an additional throws declaration instead

File: extensions/indexes/indexes-integration-tests/src/test/java/org/exist/indexing/EnforceIndexUseTest.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+import org.exist.test.ExistEmbeddedServer;
+import org.exist.test.TestConstants;
+import org.exist.util.LockException;
+import org.exist.util.MimeType;
+import org.exist.util.StringInputSource;
+import org.exist.xmldb.XmldbURI;
+import org.exist.xquery.XPathException;
+import org.exist.xquery.XQuery;
+import org.exist.xquery.XQueryContext;
+import org.exist.xquery.value.Item;
```
Comment (reinhapa) [inline extensions/indexes/indexes-integration-tests/src/test/java/org/exist/indexing/EnforceIndexUseTest.java:26]: Remove this unused import

**Review by reinhapa — CHANGES_REQUESTED** [review]

File: extensions/indexes/indexes-integration-tests/src/test/java/org/exist/indexing/EnforceIndexUseTest.java [hunk 3]
```diff
@@ -11,6 +11,7 @@
 import org.exist.security.PermissionDeniedException;
 import org.exist.storage.BrokerPool;
 import org.exist.storage.DBBroker;
+import org.exist.storage.txn.TransactionException;
```
Comment (reinhapa) [inline extensions/indexes/indexes-integration-tests/src/test/java/org/exist/indexing/EnforceIndexUseTest.java:14]: The `TransactionException` import seems to be unused and should be removed...

**Review by reinhapa — CHANGES_REQUESTED** [review]

**Review by reinhapa — APPROVED** [review]

**Comment by joewiz** [comment]
@marmoure Thank you for this PR and adding the tests!

**Review by adamretter — APPROVED** [review]
LGTM. Thanks @marmoure

**Pull request merged**